File: backend/src/main/java/com/candyshop/service/BannerService.java
package com.candyshop.service;

import com.candyshop.dto.BannerRequest;
import com.candyshop.dto.BannerResponse;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface BannerService {

    List<BannerResponse> getActiveBanners();

    List<BannerResponse> getAllBanners();

    BannerResponse getBannerById(Long id);

    BannerResponse createBanner(BannerRequest request, MultipartFile image);

    BannerResponse updateBanner(Long id, BannerRequest request, MultipartFile image);

    void deleteBanner(Long id);

    BannerResponse toggleBannerStatus(Long id, Boolean active);
}

File: frontend/src/api/bannerService.js
import axiosClient from './axiosClient';

const bannerService = {
  // Public: Active banners
  getActiveBanners: async () => {
    const response = await axiosClient.get('/api/banners');
    return response.data;
  },

  // Admin: All banners
  getAllBanners: async () => {
    const response = await axiosClient.get('/api/admin/banners');
    return response.data;
  },

  getBannerById: async (id) => {
    const response = await axiosClient.get(`/api/admin/banners/${id}`);
    return response.data;
  },

  createBanner: async (formData) => {
    const response = await axiosClient.post('/api/admin/banners', formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
    return response.data;
  },

  updateBanner: async (id, formData) => {
    const response = await axiosClient.put(`/api/admin/banners/${id}`, formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
    return response.data;
  },

  deleteBanner: async (id) => {
    const response = await axiosClient.delete(`/api/admin/banners/${id}`);
    return response.data;
  },

  toggleBannerStatus: async (id, active) => {
    const url = active !== undefined 
      ? `/api/admin/banners/${id}/status?active=${active}`
      : `/api/admin/banners/${id}/status`;
    const response = await axiosClient.put(url);
    return response.data;
  },
};

export default bannerService;
